examples/BERT/data.py

```diff
@@ -0,0 +1,79 @@
+import glob
+import torch
+import logging
+from torchtext.data.utils import get_tokenizer
+import random
+
+
+class LanguageModelingDataset(torch.utils.data.Dataset):
+    """Defines a dataset for language modeling.
+    """
+
+    def __init__(self, data, vocab):
+        """Initiate language modeling dataset.
+        """
+
+        super(LanguageModelingDataset, self).__init__()
+        self.data = data
+        self.vocab = vocab
+
+    def __getitem__(self, i):
+        return self.data[i]
+
+    def __len__(self):
+        return len(self.data)
+
+    def __iter__(self):
+        for x in self.data:
+            yield x
+
+    def get_vocab(self):
+        return self.vocab
+
+
+###################################################################
+# Set up dataset for book corpus
+###################################################################
+def BookCorpus(vocab, tokenizer=get_tokenizer("basic_english"),
```

Comment:
Maybe it's worth moving this into experimental as well?

Reply:
I was thinking about this. But the original data for BookCorpus comes from FAIR cluster.